samcli/lib/telemetry/metric.py

```diff
@@ -185,6 +202,22 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
+def _get_project_details(hook_package_id: str, template_dict: Dict) -> ProjectDetails:
+    if not hook_package_id:
+        project_type = ProjectTypes.CDK.value if is_cdk_project(template_dict) else ProjectTypes.CFN.value
+        return ProjectDetails(project_type=project_type, hook_package_id=None, hook_package_version=None)
+    hook_location = Path(INTERNAL_PACKAGES_ROOT, hook_package_id)
+    try:
+        hook_package_config = HookPackageConfig(package_dir=hook_location)
+    except InvalidHookPackageConfigException:
```

Comment:
The way we now invoke the hook (via a click option class), if it fails, seems like it won't  trigger the `track_command` method. 

So this seems like an unlikely path that will get triggered.

But we should think about how to capture that kind of failures and make sure we report them via Telemetry.

Reply:
We discussed this point offline. We generally don't send telemetry for bad click options which is a miss on our end. We will create a ticket internally to track this and follow up.